src/wp-admin/users.php

```diff
@@ -370,13 +370,17 @@
 			<li><input type="radio" id="delete_option1" name="delete_option" value="reassign" />
 				<?php
 				echo '<label for="delete_option1">' . __( 'Attribute all content to:' ) . '</label> ';
-				wp_dropdown_users(
-					array(
-						'name'    => 'reassign_user',
-						'exclude' => $userids,
-						'show'    => 'display_name_with_login',
-					)
-				);
+				$users = get_users( array( 'exclude' => $userids ) );
+					usort($users, function($a, $b) {
+						$role_a = $a->roles[0];
+						$role_b = $b->roles[0];
+						return strcmp($role_a, $role_b);
+					});
+					echo '<select name="reassign_user">';
+					foreach ( $users as $user ) {
+						echo '<option value="' . esc_attr( $user->ID ) . '"> User: ' . esc_html( $user->display_name ) . ' - rol: ' . esc_html( $user->roles[0] ) . '</option>';
```

Comment:
Missing "e" to "role".
Needs translation for "User" and "role".
Add more check to empty role 
Prefers use reset for $user->roles